plan/plan_to_pb.go

```diff
@@ -0,0 +1,110 @@
+// limitations under the License.
+
+package plan
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/distsql"
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/tablecodec"
+	"github.com/pingcap/tipb/go-tipb"
+)
+
+// ToPB implements PhysicalPlan ToPB interface.
+func (p *basePhysicalPlan) ToPB(_ context.Context) (*tipb.Executor, error) {
+	return nil, errors.Errorf("plan %s fails converts to PB", p.basePlan.id)
+}
+
+// ToPB implements PhysicalPlan ToPB interface.
+func (p *PhysicalAggregation) ToPB(ctx context.Context) (*tipb.Executor, error) {
+	sc := ctx.GetSessionVars().StmtCtx
+	client := ctx.GetClient()
+	aggExec := &tipb.Aggregation{
+		GroupBy: expression.ExpressionsToPBList(sc, p.GroupByItems, client),
+	}
+	for _, aggFunc := range p.AggFuncs {
+		aggExec.AggFunc = append(aggExec.AggFunc, expression.AggFuncToPBExpr(sc, client, aggFunc))
+	}
+	return &tipb.Executor{Tp: tipb.ExecType_TypeAggregation, Aggregation: aggExec}, nil
+}
+
+// ToPB implements PhysicalPlan ToPB interface.
+func (p *Selection) ToPB(ctx context.Context) (*tipb.Executor, error) {
+	sc := ctx.GetSessionVars().StmtCtx
+	client := ctx.GetClient()
+	selExec := &tipb.Selection{
+		Conditions: expression.ExpressionsToPBList(sc, p.Conditions, client),
+	}
+	return &tipb.Executor{Tp: tipb.ExecType_TypeSelection, Selection: selExec}, nil
+}
+
+// ToPB implements PhysicalPlan ToPB interface.
+func (p *Sort) ToPB(ctx context.Context) (*tipb.Executor, error) {
+	sc := ctx.GetSessionVars().StmtCtx
+	client := ctx.GetClient()
+	count := int64(p.ExecLimit.Count)
+	topNExec := &tipb.TopN{
+		Limit: &count,
+	}
+	for _, item := range p.ByItems {
+		topNExec.OrderBy = append(topNExec.OrderBy, expression.SortByItemToPB(sc, client, item.Expr, item.Desc))
+	}
+	return &tipb.Executor{Tp: tipb.ExecType_TypeTopN, TopN: topNExec}, nil
+}
+
+// ToPB implements PhysicalPlan ToPB interface.
+func (p *Limit) ToPB(ctx context.Context) (*tipb.Executor, error) {
+	count := int64(p.Count)
+	limitExec := &tipb.Limit{
+		Limit: &count,
+	}
+	return &tipb.Executor{Tp: tipb.ExecType_TypeLimit, Limit: limitExec}, nil
+}
+
+// ToPB implements PhysicalPlan ToPB interface.
+func (p *PhysicalTableScan) ToPB(ctx context.Context) (*tipb.Executor, error) {
+	tsExec := &tipb.TableScan{
+		TableId: p.Table.ID,
+		Columns: distsql.ColumnsToProto(p.Columns, p.Table.PKIsHandle),
+		Desc:    &p.Desc,
+	}
+	err := setPBColumnsDefaultValue(ctx, tsExec.Columns, p.Columns)
+	return &tipb.Executor{Tp: tipb.ExecType_TypeTableScan, TblScan: tsExec}, errors.Trace(err)
+}
+
+func setPBColumnsDefaultValue(ctx context.Context, pbColumns []*tipb.ColumnInfo, columns []*model.ColumnInfo) error {
+	for i, c := range columns {
+		if c.OriginDefaultValue == nil {
+			continue
+		}
+
+		sessVars := ctx.GetSessionVars()
+		originStrict := sessVars.StrictSQLMode
+		sessVars.StrictSQLMode = false
```

Comment:
why we set it false here?
It seems StrictSQLMode has no effect if c.OriginDefaultValue != nil?